src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -434,145 +434,321 @@ private static bool SkipGeneratedBranchesForAwaitForeach(List<Instruction> instr
 
             int currentIndex = instructions.BinarySearch(instruction, new InstructionByOffsetComparer());
 
-            return SkipGeneratedBranchForAwaitForeach_CheckForAsyncEnumerator(instructions, instruction, currentIndex) ||
-                   SkipGeneratedBranchForAwaitForeach_CheckIfExceptionThrown(instructions, instruction, currentIndex) ||
-                   SkipGeneratedBranchForAwaitForeach_CheckThrownExceptionType(instructions, instruction, currentIndex);
-        }
+            return CheckForAsyncEnumerator(instructions, instruction, currentIndex) ||
+                   CheckIfExceptionThrown(instructions, instruction, currentIndex) ||
+                   CheckThrownExceptionType(instructions, instruction, currentIndex);
 
-        // The pattern for the "should we stay in the loop or not?", which we don't
-        // want to skip (so we have no method to try to find it), looks like this:
-        //
-        // IL_0111: ldloca.s 4
-        // IL_0113: call instance !0 valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ValueTaskAwaiter`1<bool>::GetResult()
-        // IL_0118: brtrue IL_0047
-        //
-        // In Debug mode, there are additional things that happen in between
-        // the "call" and branch, but it's the same idea either way: branch
-        // if GetResult() returned true.
-
-        private static bool SkipGeneratedBranchForAwaitForeach_CheckForAsyncEnumerator(List<Instruction> instructions, Instruction instruction, int currentIndex)
-        {
-            // We're looking for the following pattern, which checks whether a
-            // compiler-generated field of type IAsyncEnumerator<> is null.
+
+            // The pattern for the "should we stay in the loop or not?", which we don't
+            // want to skip (so we have no method to try to find it), looks like this:
+            //
+            // IL_0111: ldloca.s 4
+            // IL_0113: call instance !0 valuetype [System.Private.CoreLib]System.Runtime.CompilerServices.ValueTaskAwaiter`1<bool>::GetResult()
+            // IL_0118: brtrue IL_0047
             //
-            // IL_012b: ldarg.0
-            // IL_012c: ldfld class [System.Private.CoreLib]System.Collections.Generic.IAsyncEnumerator`1<int32> AwaitForeachStateMachine/'<AsyncAwait>d__0'::'<>7__wrap1'
-            // IL_0131: brfalse.s IL_0196
+            // In Debug mode, there are additional things that happen in between
+            // the "call" and branch, but it's the same idea either way: branch
+            // if GetResult() returned true.
 
-            if (instruction.OpCode != OpCodes.Brfalse &&
-                instruction.OpCode != OpCodes.Brfalse_S)
+
+            bool CheckForAsyncEnumerator(List<Instruction> instructions, Instruction instruction, int currentIndex)
```

Comment:
Can you use `static` local function everywhere? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/static-local-functions

Reply:
Yep.  Will do.